app/V1Module/presenters/SubmitPresenter.php

```diff
@@ -101,6 +107,20 @@ class SubmitPresenter extends BasePresenter {
 
 
   /**
+   * Determine if given user can submit solutions to assignment.
+   * @param Assignment $assignment
+   * @param User|NULL $user
+   * @return bool
+   */
+  private function canReceiveSubmissions(Assignment $assignment, User $user = null) {
+    return $assignment->isPublic() &&
```

Comment:
I am not completely sure this is right. Take the following scenario. The teacher assigns an exercise. Once the submits begin to arrive, the teacher realizes that there is a mistake, so he/she make the assignment not public, so no more students will get wrong evaluations until the exercise is fixed. Once it is fixed, the patch needs to be tested, preferably on already submitted solutions -- i.e., the teacher needs to re-submit the solutions before making the assignment public again.

Reply:
canReceiveSubmissions() is not used during resubmits.

Reply:
OK. My mistake. Let the merging commence!